authors/apps/authentication/tests/test_reset_password.py

```diff
@@ -0,0 +1,106 @@
+import json
+
+from rest_framework.test import APIClient
+from django.test import TestCase
+from rest_framework import status
+from django.core import mail
+from .test_setup import BaseSetUp
+
+
+class PasswordTestCase(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.base = BaseSetUp()
+
+        self.login_data = {
+            'username': self.base.reg_data['username'],
+            'password': self.base.reg_data['password']
+        }
+        # login the user
+        self.login_res = self.client.post(
+            'api/users/login',
+            self.login_data,
+            format="json")
+        self.token = "self.login_res.data['Token']"
+
+        self.forgot_password = {
+            'email': 'remmy@test.com'
+        }
+
+        self.reset_password = {
+            'email': 'remmy@test.com',
+            'token': '1234567890',
+            'new_password': 'Password123'
+        }
+
+        self.renew_password = {
+            'email': 'remmy@test.com',
+            'old': 'password1',
+            'new_password': 'Password123'
+        }
+
+        self.old_password = {
+            'email': 'remmy@test.com',
+            'old': 'password1'
+        }
+
+    def test_send_forgot_password_mail(self):
+        ''' Test that checks if a reset password mail is sent'''
+        self.login_res = self.client.post(
+            'api/users/login',
+            self.login_data,
+            format="json")
+        response = self.client.post(
+            'api/users/forgot_password',
+            data=json.dumps(self.forgot_password),
+            format='json')
+
+        self.assertEqual(len(mail.outbox), 1)
+        self.assertEqual(mail.outbox[0].subject, 'Password reset link')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_reset_password(self):
+        ''' Test that checks if a user can reset password '''
+        self.login_res = self.client.post(
+            'api/users/login',
+            self.login_data,
+            format="json")
+        response = self.client.put(
+            'api/users/forgot_password',
+            data=json.dumps(self.reset_password),
+            format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(json.loads(response.data)["message"],
+                         "Password updated successfully, you may now login")
+
+    def test_renew_password(self):
+        ''' Test that checks if a logged in user can renew password '''
+
+        response = self.client.put(
+            'api/users/forgot_password',
+            data=json.dumps(self.renew_password),
```

Comment:
Should we have this endpoint for renew password?

Reply:
Also nothing on this test shows that the user has logged in first before trying to reset password.